dev/validate_bom__test.py

```diff
@@ -735,6 +818,13 @@ def validate_test_requirements(self, test_name, spec, metric_labels):
       raise_and_log_error(
           UnexpectedError('Test "{name}" is missing an "api" spec.'.format(
               name=test_name)))
+
+    if spec['api'] in self.__service_endpoints:
+      # No need to wait on service if we are using the service endpoint.
```

Comment:
just because the api is for consideration does not mean that an endpoint is defined. You either need to check here, or remove from teh map the unconfigured endpoints.